### PR TITLE
Cleanup newfeat file

### PR DESCRIPTION
**Pull request by jabranham**
Ensure first line is a sentence and ends with a period, group
fontification stuff under a heading, move notable bug fixes to the
bottom, and various minor changes.

### CODE REVIEW

**Comment by jabranham** [comment]
This part:

```
@item @ESS{[R]}: We have improved fontification of keywords so they
better reflect the semantics of the R language. @code{ess-R-keywords}
now only contains words reserved by the R parser.
@code{ess-R-control-flow-keywords} contains words of base functions that
cause non-contiguous control flow, such as @code{return()} and
@code{stop()}. It includes the following variables that were previously
not fontified: @code{on.exit()}, @code{tryCatch()},
@code{withRestarts()}, @code{invokeRestart()}, @code{recover()} and
@code{browser()}
 Finally, @code{ess-R-signal-keywords} contains functions part of the
condition system that only potentially impact control flow:
@code{message()}, @code{warning()} (moved from @code{ess-R-keywords}),
@code{signalCondition()} and @code{withCallingHandlers()}. These
keywords inherit from @code{ess-modifiers-face} (the face used for
@code{library()} etc).
```

doesn't seem true anymore but I admit I've gotten a little lost with all the recent fontification changes. Would be nice @lionel- if you could confirm removing it is OK.

File: doc/newfeat.texi [hunk 1]
```diff
@@ -128,58 +86,91 @@ It provides similar functionality but for any keybinding, not just `_`.
...
+ESS (i.e. in the @code{:config} section for @code{use-package} users).
+
+@item @ESS{[R]}: Fontification of roxygen @code{@@param} keywords now supports comma-separated parameters.
+
+@item @ESS{[R]}: Certain keywords are only fontified if followed by a parenthesis.
+Function-like keywords such as @code{if ()} or
+@code{stop()} are no longer fontified as keyword if not followed by an
+opening parenthesis. The same holds for search path modifiers like
+@code{library()} or @code{require()}. This feature is only available
+in Emacs >= 25.
```
Comment (lionel-) [inline doc/newfeat.texi]: >  This feature is only available in Emacs >= 25.

This is no longer true.

**Comment by lionel-** [comment]
Right, it should become something like:

Many new keywords were added to `ess-R-keywords` (`on.exit()`, `browser()`, ...) and `ess-R-modifiers` (`options()`, `rm()`, `signalCondition()`, ...).

File: doc/newfeat.texi (hunk 1 above)
Comment (jabranham) [reply doc/newfeat.texi]: Thanks, removed.

**Comment by jabranham** [comment]
Thanks. I added:

```
@item Many new keywords were added to @code{ess-R-keywords} and @code{ess-R-modifiers}.
See the documentation for details.
```

and pushed.

File: doc/newfeat.texi [hunk 2]
```diff
@@ -4,117 +4,75 @@
...
-@code{prog-mode} or @code{text-mode}. If you want to restore the old
-behavior and activate the package mode in all buffers (that have a
-@code{default-directory} that is part of a package path), set
-@code{ess-r-package-auto-activate} to @code{t}.
-
-@item ESS now provides support for flymake in R buffers for Emacs 26 and newer.
-Users need to install the @code{lintr} package (available on CRAN) to
-use it. Customizable options include @code{ess-use-flymake},
-@code{ess-r-flymake-linters}, and @code{ess-r-flymake-lintr-cache}.
+@item @ESS{[R]}: The package development minor mode only activates in file-visiting buffers.
```
Comment (vspinu) [inline doc/newfeat.texi]: This one was reverted and is no longer true.

File: doc/newfeat.texi [hunk 3]
```diff
@@ -4,117 +4,75 @@
...
+@item ESS now respects Emacs conventions for keybindings.
+This means that The @code{C-c [letter]} bindings have been
+removed. This affects @code{C-c h}, which was bound to
+@code{ess-eval-line-and-step-invisibly} in @code{sas-mode-local-map};
+@code{C-c f}, which was bound to @code{ess-insert-function-outline} in
+@code{ess-add-MM-keys}; and @code{C-c h}, which was bound to
+@code{ess-handy-commands} in @code{Rd-mode-map},
+@code{ess-noweb-minor-mode-map}, and @code{ess-help-mode-map}
+
+@item @ESS{[R]}: @code{ess-r-package-use-dir} now works with any mode.
```
Comment (vspinu) [inline doc/newfeat.texi]: I don't understand this entry. Can we make it more user friendly without mentioning `ess-r-package-use-dir` which is an internal function?

File: doc/newfeat.texi [hunk 4]
```diff
@@ -4,117 +4,75 @@
...
-@code{ess-noweb-minor-mode-map}, and @code{ess-help-mode-map}
-
-@item prettify-symbols-mode no longer breaks indentation
-This is accomplished by having the pretty symbols occupy the same
-number of characters as their non-pretty cousins.  You may customize
-the new variable @code{ess-r-prettify-symbols} to control this
-behavior.
-
-@item The @code{ess-r-args.el} library has obsoleted and will be removed in a future release.
+@item The @code{ess-r-args.el} library has obsoleted and will be removed in the next release.
```
Comment (vspinu) [inline doc/newfeat.texi]: Can we make all the obsoleted/removed items together?

File: doc/newfeat.texi [hunk 5]
```diff
@@ -4,117 +4,75 @@
...
+@code{ess-add-MM-keys}; and @code{C-c h}, which was bound to
+@code{ess-handy-commands} in @code{Rd-mode-map},
+@code{ess-noweb-minor-mode-map}, and @code{ess-help-mode-map}
+
+@item @ESS{[R]}: @code{ess-r-package-use-dir} now works with any mode.
+This sets the working directory to the root of the current package
+including for example C or C++ files within @code{/src}).
+
 @item @ESS{[R]}: Long + + prompts in the inferior no longer offset output.
 New option @code{strip} for @code{inferior-ess-replace-long+} strips the
```
Comment (vspinu) [inline doc/newfeat.texi]: This one should be split in a separate item I believe.

File: doc/newfeat.texi (hunk 3 above)
Comment (jabranham) [reply doc/newfeat.texi]: `ess-r-package-use-dir` isn't internal. It's interactive. I don't think this entry is confusing; what do you not like about it?

File: doc/newfeat.texi (hunk 2 above)
Comment (jabranham) [reply doc/newfeat.texi]: OK, removed.

File: doc/newfeat.texi (hunk 4 above)
Comment (jabranham) [reply doc/newfeat.texi]: Yes, done

File: doc/newfeat.texi (hunk 5 above)
Comment (jabranham) [reply doc/newfeat.texi]: OK, done

File: doc/newfeat.texi (hunk 3 above)
Comment (vspinu) [reply doc/newfeat.texi]: Nevermind. I didn't know we have this function. Thought it's something else.

Comment (jabranham) [reply doc/newfeat.texi]: OK, merged

**Pull request merged**